app/controllers/application_controller.rb

```diff
@@ -2,4 +2,22 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :configure_devise_permitted_parameters, if: :devise_controller?
+
+   protected
+
+  def configure_devise_permitted_parameters
+    registration_params = [:username, :email, :password, :password_confirmation]
+
+    if params[:action] == 'update'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.